src/contracts/BatchExchangeContract.ts

```diff
@@ -0,0 +1,166 @@
+
+export interface Order {
+  buyToken: string
+  sellToken: string
+  validFrom: string
+  validUntil: string
+  priceNumerator: string
+  priceDenominator: string
+  usedAmount: string
+}
+
+export interface SolutionData {
+  batchId: string
+  solutionSubmitter: string
+  feeReward: string
+  objectiveValue: string
+}
+
+export interface OrderCancelation {
+  owner: string
+  id: string
+}
+
+export interface Deposit {
+  user: string
+  token: string
+  amount: string
+  stateIndex: string
+}
+
+export interface WithdrawRequest {
+  user: string
+  token: string
+  amount: string
+  stateIndex: string
+}
+
+export interface Withdraw {
+  user: string
+  token: string
+  amount: string
+}
+
+export interface OrderPlacement {
+  owner: string
+  buyToken: string
+  sellToken: string
+  validFrom: string
+  validUntil: string
+  priceNumerator: string
+  priceDenominator: string
+}
+
+export interface BatchExchangeContract extends Contract {
+  // constructor(jsonInterface: any[], address?: string, options?: ContractOptions)
+  // constructor (jsonInterface: AbiItem[], address?: string, options?: ContractOptions): BatchExchange;
+
+  clone(): BatchExchangeContract
+
+  methods: {
+    getSecondsRemainingInBatch(): TransactionObject<string>
+
+    feeDenominator(): TransactionObject<string>
+
+    getPendingWithdrawAmount(user: string, token: string): TransactionObject<string>
+
+    requestWithdraw(token: string, amount: number | string): TransactionObject<void>
+
+    getPendingDepositAmount(user: string, token: string): TransactionObject<string>
+
+    deposit(token: string, amount: number | string): TransactionObject<void>
+
+    getPendingWithdrawBatchNumber(user: string, token: string): TransactionObject<string>
+
+    TOKEN_ADDITION_FEE_IN_OWL(): TransactionObject<string>
+
+    feeToken(): TransactionObject<string>
+
+    currentPrices(arg0: number | string): TransactionObject<string>
+
+    orders(arg0: string, arg1: number | string): TransactionObject<Order>
+
+    numTokens(): TransactionObject<string>
+
+    lastCreditBatchId(arg0: string, arg1: string): TransactionObject<string>
+
+    latestSolution(): TransactionObject<SolutionData>
+
+    getBalance(user: string, token: string): TransactionObject<string>
+
+    BATCH_TIME(): TransactionObject<string>
+
+    getCurrentBatchId(): TransactionObject<string>
+
+    requestFutureWithdraw(token: string, amount: number | string, batchId: number | string): TransactionObject<void>
+
+    hasValidWithdrawRequest(user: string, token: string): TransactionObject<boolean>
+
+    MAX_TOKENS(): TransactionObject<string>
+
+    getPendingDepositBatchNumber(user: string, token: string): TransactionObject<string>
+
+    withdraw(user: string, token: string): TransactionObject<void>
+
+    MAX_TOUCHED_ORDERS(): TransactionObject<string>
+
+    addToken(token: string): TransactionObject<void>
+
+    placeValidFromOrder(
+      buyToken: number | string,
```

Comment:
could be better to create a type from `number | string` - your call, I'm fine either way

Reply:
As we generate or will generate these types with typechain, it's not up to us